Bindings/Java/OpenSimJNI/OpenSimContext.cpp

```diff
@@ -243,30 +233,26 @@ void OpenSimContext::setLocation(PathPoint& mp, int i, double d) {
     SimTK::Vec3 loc = mp.getLocation(*_configState);
     loc[i] = d;
     mp.setLocation(loc);
-    _configState->invalidateAll(SimTK::Stage::Position);
-    _model->getMultibodySystem().realize(*_configState, SimTK::Stage::Position);
+    recreateSystemKeepStage();
 }
 
 void OpenSimContext::setEndPoint(PathWrap& mw, int newEndPt) {
     mw.setEndPoint(*_configState, newEndPt );
-  _configState->invalidateAll(SimTK::Stage::Position);
-  _model->getMultibodySystem().realize(*_configState, SimTK::Stage::Position);
+    recreateSystemKeepStage();
     return;
 }
 
 void OpenSimContext::addPathPoint(GeometryPath& p, int menuChoice, PhysicalFrame&  body) {
     p.addPathPoint(*_configState, menuChoice, body );
-  _configState->invalidateAll(SimTK::Stage::Position);
-  _model->getMultibodySystem().realize(*_configState, SimTK::Stage::Position);
-  p.updateGeometry(*_configState);
+    recreateSystemKeepStage();
+    p.updateGeometry(*_configState);
     return;
 }
 
 bool OpenSimContext::deletePathPoint(GeometryPath& p, int menuChoice) {
     bool ret = p.deletePathPoint(*_configState, menuChoice );
-  _configState->invalidateAll(SimTK::Stage::Position);
-  _model->getMultibodySystem().realize(*_configState, SimTK::Stage::Position);
-  return ret;
+    recreateSystemKeepStage();
```

Comment:
Nice catch @tkuchida, it was more of blanket code transformation to replace the invalidate/realize call pair with recreateSystemKeepStage but indeed that is not necessary in this case :+1: Will make the change, thank you.